src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -223,14 +223,18 @@ CommandsRegistry.registerCommand({
 
 		for (let editor of editors) {
 			if (editor instanceof NotebookInput) {
-				let model: INotebookModel = editor.notebookModel;
-				if (model?.providerId === JUPYTER_PROVIDER_ID) {
-					let jupyterNotebookManager: IExecuteManager = model.executeManagers.find(x => x.providerId === JUPYTER_PROVIDER_ID);
-					await jupyterNotebookManager.sessionManager.shutdownAll();
-					jupyterNotebookManager.sessionManager.dispose();
-					await jupyterNotebookManager.serverManager.stopServer();
-					return;
-				}
+				// Wait for NotebookInput.resolve() to be called to ensure the _model field in NotebookInput
+				// is not undefined when trying to get the notebookModel.
+				editor.modelResolved.then(async () => {
+					let model: INotebookModel = (<NotebookInput>editor).notebookModel;
```

Comment:
So the behavior is changing slightly in that now we'll go through all the editors and kick off the shutdowns async'ly. Could this result in some race condition errors being thrown if multiple editors share a server?

You may want to just await editor.modelResolved so the behavior is the same as previously where at least we'd fully shut down a server before moving on to the next.